packages/picasso.js/src/core/math/intersection.js

```diff
@@ -17,3 +32,40 @@ export function isCollinear(p1, p2, p3, e = EPSILON) {
   const t = Math.abs((p1.x * (p2.y - p3.y)) + (p2.x * (p3.y - p1.y)) + (p3.x * (p1.y - p2.y)));
   return t < e;
 }
+
+/**
+ * Check if the child rect is located entirely within it's parent rect. If no settings object is provided,
+ * it tries to check if the child rect fits assuming ltr mode and orientation at middle of the child rect.
+ * Returns true if child rect fits inside the parent rect, otherwise false.
+ * @param {rect} childRect - A rect describing the position(x,y), width and height of label
+ * @param {rect} parentRect - A rect describing the position(x,y), width and height of parent rect
+ * @param {object} settings - An object containing alignment(rtl or ltr) of childRect and orientation (optional)
+ * @ignore
+ */
+export function rectContainsRect(childRect, parentRect, settings) {
```

Comment:
Please move this function and the other utility functions back to the labels namespace. As this is not a generic function for rectContainsRect. That would look something like this:
```
function rectContainsRect(a, b) {
    return a.x >= b.x &&
        a.x + a.width <= b.x + b.width &&
        a.y >= b.y &&
        a.y + a.height <= b.y + b.height;
}
```

Also I wouldn't bother with the type checking and throwing errors. As we don't have that else where in picasso.